tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,213 @@
+                  metrics=['accuracy'])
+    run = mlflow.start_run()
+    model.fit(data, labels, epochs=10)
+    mlflow.end_run()
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_keras_random_data_run_instance(tf_keras_random_data):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_keras_random_data)
+    return run
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_epoch_acc(get_tf_keras_random_data_run_instance):
+    assert 'epoch_acc' in get_tf_keras_random_data_run_instance.data.metrics
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_epoch_loss(get_tf_keras_random_data_run_instance):
+    assert 'epoch_loss' in get_tf_keras_random_data_run_instance.data.metrics
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_optimizer_name(get_tf_keras_random_data_run_instance):
+    assert 'optimizer_name' in get_tf_keras_random_data_run_instance.data.params
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_model_summary(get_tf_keras_random_data_run_instance):
+    assert 'summary' in get_tf_keras_random_data_run_instance.data.tags
+    assert get_tf_keras_random_data_run_instance.data.tags['summary'] is not ''
+
+
+@pytest.mark.large
+def test_tf_keras_autolog_saves_model(get_tf_keras_random_data_run_instance):
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(get_tf_keras_random_data_run_instance.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert 'model' in artifacts
+    assert 'tensorboard_logs' in artifacts
+
+
+@pytest.fixture
+def tf_core_random_tensors():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=1)
+    run = mlflow.start_run()
+    sess = tf.Session()
+
+    a = tf.constant(3.0, dtype=tf.float32)
+    b = tf.constant(4.0)  # also tf.float32 implicitly
+    total = a + b
+    tf.summary.scalar('a', a)
+    tf.summary.scalar('b', b)
+    merged = tf.summary.merge_all()
+    dir = tempfile.mkdtemp()
+    writer = tf.summary.FileWriter(dir, sess.graph)
+    with sess.as_default():
+        summary, _ = sess.run([merged, total])
+
+    writer.add_summary(summary)
+
+    shutil.rmtree(dir)
+
+    writer.close()
+
+    mlflow.end_run()
+
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_core_random_tensors_run_instance(tf_core_random_tensors):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_core_random_tensors)
+    return run
+
+
+@pytest.mark.small
+def test_tf_core_autolog_logs_scalars(get_tf_core_random_tensors_run_instance):
+    assert 'a' in get_tf_core_random_tensors_run_instance.data.metrics
+    assert get_tf_core_random_tensors_run_instance.data.metrics['a'] == 3.0
+    assert 'b' in get_tf_core_random_tensors_run_instance.data.metrics
+    assert get_tf_core_random_tensors_run_instance.data.metrics['b'] == 4.0
+
+
+@pytest.fixture
+def tf_estimator_random_data():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=1)
+    run = mlflow.start_run()
```

Comment:
Nit: I'd use `with mlflow.start_run()` so that the run is automatically ended even if we fail an assert/throw within the test, otherwise we'll risk leaking an active run.

Reply:
Good catch